packages/react/react-query/package.json

```diff
@@ -11,7 +15,7 @@
   },
   "main": "./src/index.ts",
   "scripts": {
-    "build": "rm -rf dist esm && tsc --declaration --emitDeclarationOnly --declarationDir dist && rollup -c rollup.config.js",
+    "build": "rm -rf dist esm && tsc --declaration --emitDeclarationOnly --declarationDir dist && rollup -c rollup.config.js && pnpify tsd",
```

Comment:
### Check it is fine please
I chained ```pnpify tsd``` after original build
It will make error in process of build if there is error in tsd code.

Is it fine for you?

Reply:
Why not running tests in the `test` script, rather than `build`? We always check if the `test` passes before a PR is merged.

Reply:
@raon0211 
I want to split it(`pnpify tsd`) from `yarn build` too.
but cannot find any code that execute just only `test` before a PR is merged.
Where can i check that code?

I found only circleci using [jest in .circleci/config.yml](https://github.com/toss/slash/blob/main/.circleci/config.yml#L112-L161), not yarn test.
Isn't `test` just used as just naming, right?

Is there any setting executing `yarn test` not visible for me?
or Could you help me check it out please?

Reply:
@HoseungJang Okay.
I removed tsd in script:`build` and [I added a script:`tsd`](https://github.com/toss/slash/pull/99/commits/fb022afa27c97b616ab4d249f3e64b8823672afc) seperately.
It is intended to be easy to use whether you know tsd needs pnpify on yarn berry project or not.

```json
  "scripts": {
    "build": "rm -rf dist esm && tsc --declaration --emitDeclarationOnly --declarationDir dist && rollup -c rollup.config.js",
    "prepack": "yarn build",
    "test": "echo \"no test specified.\"",
    "tsd": "pnpify tsd", // added: to use tsd on yarn berry
    "typecheck": "tsc --noEmit"
  },
```

If anyone have used script `build`, it will make definition files so you can use `tsd` after.
How about next new Pull Request for doing better thing that you said after merging this Pull Request?

Reply:
We are discussing about this. Please wait a bit 🙏

Reply:
Seperating tsd script and adding that in prepack lifecycle seems good to me.
But like you said, it would be better to do it in the new PR.
This seems fine to merge.

```json
  "scripts": {
    "prepack": "yarn build && yarn tsd",
    "tsd": "pnpify tsd"
  },
```

Reply:
I fixed scripts as what you comment in [a42814c](https://github.com/toss/slash/pull/99/commits/a42814c5b136452fe8ea292ea10c1b873b76dedb)

thanks for your guide :)